pkg/controller/velerobackup/velerobackup_controller.go

```diff
@@ -151,5 +182,75 @@ func createVeleroBackupObject(client client.Client, namespace string) error {
 		},
 	}
 
-	return client.Create(context.TODO(), backup)
+	return r.Create(context.TODO(), backup)
+}
+
+func (r *ReconcileBackup) getHiveObjectsInNamespace(namespace string, filter filterFunc) ([]*hiveObject, error) {
+	nsLogger := addNamespaceLoggerFields(r.logger, namespace)
+
+	filteredHiveObjects := []*hiveObject{}
+
+	allNSObjs := []runtime.Object{}
+	for _, t := range typesToList {
+		listObj := t.DeepCopyObject()
+		if err := r.List(context.TODO(), listObj, client.InNamespace(namespace)); err != nil {
+			nsLogger.WithError(err).Errorf("Couldn't get list of %s", t.GetObjectKind().GroupVersionKind().Kind)
+			return nil, err
+		}
+		list, err := meta.ExtractList(listObj)
+		if err != nil {
+			r.logger.WithError(err).Error("Failed extracting a list from results.")
+			return nil, err
+		}
+
+		for _, nsObj := range list {
+			hobj, err := newHiveObject(nsObj, r.logger)
+			if err != nil {
+				return nil, err
+			}
+
+			if filter(hobj) {
+				filteredHiveObjects = append(filteredHiveObjects, hobj)
+			}
+
+		}
+
+		allNSObjs = append(allNSObjs, list...)
+	}
+
+	return filteredHiveObjects, nil
+}
+
+func filterAllObjects(hobj *hiveObject) bool {
+	return true
+}
+
+func filterOnlyModifiedObjects(hobj *hiveObject) bool {
+	return hobj.hasChanged()
+}
+
+func (r *ReconcileBackup) updateHiveObjectsLastBackupChecksum(hobjs []*hiveObject) error {
+	errors := []error{}
+
+	for _, hobj := range hobjs {
+		hobjLogger := hobj.addLoggerFields()
+
+		hobj.setChecksumAnnotation()
+		obj := hobj.object
+
+		err := r.Update(context.TODO(), obj)
+		if err != nil {
+			hobjLogger.WithError(err).Error("Failed update.")
+			errors = append(errors, err)
+			continue // It errored, so jump to next loop.
```

Comment:
@staebler as it sits now, we're not re-queuing the namespace immediately (we just log the error).

Do you still think we should return on first update failure?

Reply:
The controller does not explicitly re-queue the namespace. But the resources that the controller is watching will have changed, which will cause the namespace to be re-queued.

That assumes that there was success in setting the annotation for at least one resource. In the scenario where the update of the first resource fails, then the namespace will not be re-queued. That leaves open a small possibility of error in the scenario where a resource is updated, is backed up, fails to get its annotation set, and is then reverted back to its original state. Then the backup controller will think that the backup contains the original state of the resource instead of the intermediate state so will not trigger a new backup.